src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/CSharp/ComponentTagHelperDescriptorProvider.cs

```diff
@@ -115,8 +115,10 @@ protected override void Collect(ISymbol symbol, ICollection<TagHelperDescriptor>
 
             if (fullyQualified)
             {
-                var containingNamespace = type.ContainingNamespace.ToDisplayString();
-                var fullName = $"{containingNamespace}.{type.Name}";
+                var containingNamespace = type.ContainingNamespace.ToDisplayString(SymbolExtensions.FullNameTypeDisplayFormat);
+                var fullName = string.IsNullOrEmpty(containingNamespace)
+                    ? type.Name
+                    : $"{containingNamespace}.{type.Name}";
```

Comment:
Would prefer if we were more direct with the condition here:
```suggestion
                var fullName = type.ContainingNamespace.IsGlobalNamespace
                    ? type.Name
                    : $"{type.ContainingNamespace.ToDisplayString(SymbolExtensions.FullNameTypeDisplayFormat)}.{type.Name}";
```

Reply:
Also, is it possible to have a component in a nested type? If so, will this handle that correctly?

Reply:
> Also, is it possible to have a component in a nested type? If so, will this handle that correctly?

I don't think it's possible. (You've already asked me this previously in https://github.com/dotnet/razor/pull/9689#discussion_r1424387085 :D)

In any case, the previous code did `"{namespace}.{typeName}"` and I've just changed it to omit the `"{namespace}."` prefix when needed, so nothing should change at this front, right?